core/deployment/src/main/java/io/quarkus/deployment/index/IndexingUtil.java

```diff
@@ -61,7 +62,11 @@ public static Index indexJar(File file, Set<String> removed) throws IOException
                                 file);
                         return indexJar(jarFile, removed);
                     } else {
-                        return reader.read();
+                        try {
+                            return reader.read();
+                        } catch (UnsupportedVersion e) {
+                            throw new UnsupportedVersion("Can't read Jandex index from " + file + ": " + e.getMessage());
```

Comment:
Hm, this way you swallow the info about supported index versions, or?

Reply:
I think that info is in the `e.getMessage()`?

Reply:
Ah, of course! Stupid me!!!